munch/__init__.py

```diff
@@ -397,13 +397,41 @@ def munchify(x, factory=Munch):
 
         nb. As dicts are not hashable, they cannot be nested in sets/frozensets.
     """
+    return cycle_munchify(x, factory, dict())
+
+# Munchify x, use seen to track object cycles
+def cycle_munchify(x, factory, seen):
+    try:
+        return seen[id(x)]
+    except KeyError:
+        pass
+
+    m = seen[id(x)] = cycle_premunchify(x, factory, seen)
+    cycle_postmunchify(m, x, factory, seen)
+    return m
+
+def cycle_premunchify(x, factory, seen):
```

Comment:
Assuming these are not intended for outside use, please prefix with `_` to prevent accidental importing